lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/dataset/LightGBMDataset.scala

```diff
@@ -175,14 +175,15 @@ class LightGBMDataset(val datasetPtr: SWIGTYPE_p_void) extends AutoCloseable {
     addIntField(groupCardinality, "group", groupCardinality.length)
   }
 
-  def setFeatureNames(featureNamesOpt: Option[Array[String]], numCols: Int): Unit = {
+  def setFeatureNames(featureNamesOpt: Option[Array[String]], numCols: Int): LightGBMDataset = {
     // Add in slot names if they exist
     featureNamesOpt.foreach { featureNamesVal =>
```

Comment:
Since this code is interested in non-empty feature names, let's filter out them first before using them. That would make the intent of the code more explicit.

```scala
featureNamesOpt.filter(_.nonEmpty).foreach { featureNamesVal =>
  ...
}
```

WDYT?

Reply:
The foreach is applied to the Option[Array[string]], so it it basically used in place of "if featureNameOpt.HasValue do this". The "nonEmpty" is applied to the array itself, looking for > 0 elements.

The suggested code above would change the logic.

This is another case of readability using "foreach" to mean "If Option[X] has value do this".  I agree it reads bad. I blame Mark for teaching me this. :)  Is there a better way to do this without using the foreach and without using an "if"?

Reply:
Let's leave it as is. I'll see if I'm of more help next time I see it 😉